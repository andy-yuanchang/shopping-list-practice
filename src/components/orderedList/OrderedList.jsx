import React from 'react';
import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';

import './OrderedList.less';

function OrderedList(props) {
  const orederedList = useSelector((state) => state.order.list);

  const renderItemList = (itemList) => (
    <>
      {
        itemList.map((item, index) => (
          <div className="item" key={index}>
            {`${item.storeName} ${item.name} x ${item.count}`}
          </div>
        ))
      }
    </>
  );

  return (
    <div className="ordered-list">
      {
        orederedList.map((order, index) => (
          <div className="order-item" key={index}>
            {renderItemList(order.itemList)}
          </div>
        ))
      }
    </div>
  );
}

OrderedList.propTypes = {
  orderedList: PropTypes.arrayOf(PropTypes.object),
};

export default OrderedList;
