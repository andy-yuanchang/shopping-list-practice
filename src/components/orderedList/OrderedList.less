@import '../generalAttribute.less';

.ordered-list {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 2rem;
  flex-wrap: wrap;
  > * {
    flex-basis: 100%;
  }
  counter-reset: order-number;
  .title {
    text-align: center;
  }
  .order-item {
    counter-increment: order-number;
    .box-shadow-style();
    &::before {
      content: 'Order ' counter(order-number);
      border-bottom: 1px solid @font-color;
    }
    padding: 1rem;
    .item {
      padding: 1rem;
    }
  }
}