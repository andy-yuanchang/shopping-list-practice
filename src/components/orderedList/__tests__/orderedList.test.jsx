import React from 'react';
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import { render, cleanup } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import OrderedList from '../OrderedList';

afterEach(cleanup);
const mockStore = configureMockStore();

function TestOrderedList({ store }) {
  return (
    <Provider store={store}>
      <OrderedList />
    </Provider>
  );
}

describe('OrderedList', () => {
  it('default view', () => {
    const store = mockStore({
      order: {
        list: [{
          itemList: [{
            storeName: 'kebuke',
            name: 'bubble tea',
            count: '5',
          }],
        }, {
          itemList: [{
            storeName: 'i lan',
            name: 'rumen',
            count: '2',
          }],
        }, {
          itemList: [{
            storeName: '7-11',
            name: 'coke',
            count: '15',
          }],
        }, {
          itemList: [{
            storeName: 'family mart',
            name: 'beer',
            count: '24',
          }],
        }],
      },
    });
    const orderedList = TestRenderer.create(
      <TestOrderedList store={store} />,
    );
    expect(orderedList.toJSON()).toMatchSnapshot();
  });

  it('empty view', () => {
    const store = mockStore({ order: { list: [] } });
    const orderedList = TestRenderer.create(
      <TestOrderedList store={store} />,
    );
    expect(orderedList.toJSON()).toMatchSnapshot();
  });
});
