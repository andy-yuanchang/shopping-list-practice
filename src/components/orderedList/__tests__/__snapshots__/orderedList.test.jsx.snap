// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderedList default view 1`] = `
<div
  className="ordered-list"
>
  <div
    className="order-item"
  >
    <div
      className="item"
    >
      kebuke bubble tea x 5
    </div>
  </div>
  <div
    className="order-item"
  >
    <div
      className="item"
    >
      i lan rumen x 2
    </div>
  </div>
  <div
    className="order-item"
  >
    <div
      className="item"
    >
      7-11 coke x 15
    </div>
  </div>
  <div
    className="order-item"
  >
    <div
      className="item"
    >
      family mart beer x 24
    </div>
  </div>
</div>
`;

exports[`OrderedList empty view 1`] = `
<div
  className="ordered-list"
/>
`;
